sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -4589,6 +4589,20 @@ object SQLConf {
       .stringConf
       .createWithDefault("versionAsOf")
 
+  val RELEASE_ANTLR_CACHE_AFTER_PARSE =
+    buildConf("spark.sql.releaseAntlrCacheAfterParse")
+      .doc("When true, release the ANTLR cache after parsing a SQL query. ANTLR parsers retain a " +
+        "DFA cache designed to speed up parsing future input. However, there is no limit to how " +
+        "large this cache can become and parsing large SQL statements can lead to an " +
+        "accumulation of objects in the cache that are unlikely to be reused.  This can deplete " +
+        "the available heap on the driver, leading to high GC overhead and evenatually OOMs. Set " +
+        "this to `true` to release the cache after every SQL statement.  On a long-lived driver, " +
+        "it may be beneficial to periodically set this to `true`, run a query to clear the " +
+        "cache, and revert it to `false`.")
+      .version("4.0.0")
+      .booleanConf
+      .createWithDefault(false)
```

Comment:
Since this is `false`, we preserve the existing behavior. Did I understand correctly?

Reply:
Yes